core/src/main/scala/com/yahoo/maha/core/query/presto/PrestoQueryGenerator.scala

```diff
@@ -491,15 +491,24 @@ class PrestoQueryGenerator(partitionColumnRenderer:PartitionColumnRenderer, udfS
       } else {
         "LEFT OUTER JOIN"
       }
-      // pkColName ?? alias ?? cc3_id
+
+      // Columns on which join is done must be of the same type. Add explicit CAST otherwise.
+      val joinCondition = {
+        if (fkCol.dataType.getClass.equals(dimBundle.dim.columnsByNameMap(pkColName).dataType.getClass)) {
+          s"$factViewAlias.$renderedFactFk = $dimAlias.${dimAlias}_$pkColName"
+        } else {
+          s"CAST($factViewAlias.$renderedFactFk AS VARCHAR) = CAST($dimAlias.${dimAlias}_$pkColName AS VARCHAR)"
```

Comment:
I am not sure how will it affect performance as it will cast all the keys in right and left table

Reply:
Checked with a few queries in prod, performance seems same.